db/schema_applier.cc

```diff
@@ -785,8 +785,31 @@ static future<> do_merge_schema(distributed<service::storage_proxy>& proxy, shar
         slogger.debug("Affected tables for keyspace {}: {}", keyspace_name, sel.tables);
     }
 
+    auto merge_keyspace_schemas = [&](auto &&keyspace, auto &&scylla_keyspace) {
+        if (!keyspace.empty() && !scylla_keyspace.empty()) {
+            for (const auto& ks_iter : keyspace) {
+                const auto &ks_name = ks_iter.first;
+                auto ks_rows = keyspace[ks_name]->rows();
+                auto ks_scylla_rows = scylla_keyspace[ks_name]->rows();
+                typeof(ks_rows) out;
+                for (auto &e: ks_rows) {
+                    out.push_back(e);
+                }
+                for (auto &e: ks_scylla_rows) {
+                    out.push_back(e);
+                }
+
+                auto keyspace_merged = make_lw_shared<query::result_set>(keyspace.begin()->second->schema(),
```

Comment:
well, this probably won't work, because you end up with a `result_set` that uses certain schema, but some rows in this set are coming from different schema. (`system.keyspaces` vs `system.scylla_keyspaces`)

Any calculations you do on such result_set will probably be incorrect

so it looks like we'll have to deal with two result_sets: one containing the rows from `system.keyspaces` and the other from `system.scylla_keyspaces`

Reply:
The CI is green, I'm resolving the comment - unless you meant something else, then please reopen it.